request/parser.go

```diff
@@ -1,10 +1,16 @@
-// Package request provides a struct for setting and retrieving relay
-// request details from the context during the relay request lifecycle.
+// The responsibility of the `request` package is to extract the service ID and find the target service's corresponding QoS instance.
+// See: https://github.com/buildwithgrove/path/blob/e0067eb0f9ab0956127c952980b09909a795b300/gateway/gateway.go#L52C2-L52C45
+//
+// Request package decides how the requested service is referenced by the user (currently: subdomain of the HTTP request).
+//
+// Processing should fail here only if:
+// A) No service is provided - Bad Request
+// B) The provided service is not found/configured for the gateway instance - Not Found
```

Comment:
Let's make a small edit.

My main concern here is that it's very "what the requests package is for right now" rather than what it'll be for in the future. I've found that engineers never go back and update package level docs.

```suggestion
// Request package is responsible for parsing and forwarding user requests.
// It is not responsible for QoS, authorization, etc...
//
// For example, Processing should fail here only if:
// A) No service is provided - Bad Request
// B) The provided service is not found/configured for the gateway instance - Not Found
//
// The responsibility of the `request` package is to extract the service ID and find the target service's corresponding QoS instance.
// See: https://github.com/buildwithgrove/path/blob/e0067eb0f9ab0956127c952980b09909a795b300/gateway/gateway.go#L52C2-L52C45
```